src/azure-cli-core/azure/cli/core/profiles/_shared.py

```diff
@@ -118,29 +118,39 @@ def get_api_version(api_profile, resource_type):
 @total_ordering  # pylint: disable=too-few-public-methods
 class _DateAPIFormat(object):
     """ Class to support comparisons for API versions in
-        YYYY-MM-DD or YYYY-MM-DD-preview format
+        YYYY-MM-DD or YYYY-MM-DD-preview format. A special case is made for 'latest'.
     """
 
     def __init__(self, api_version_str):
         try:
-            if 'preview' in api_version_str:
-                yyyy, mm, dd, _ = api_version_str.split('-')
-                self.preview = True
+            self.latest = self.preview = False
+            self.yyyy = self.mm = self.dd = None
+            if api_version_str == 'latest':
+                self.latest = True
             else:
-                yyyy, mm, dd = api_version_str.split('-')
-                self.preview = False
-            self.yyyy = int(yyyy)
-            self.mm = int(mm)
-            self.dd = int(dd)
+                if 'preview' in api_version_str:
+                    yyyy, mm, dd, _ = api_version_str.split('-')
+                    self.preview = True
+                else:
+                    yyyy, mm, dd = api_version_str.split('-')
+                self.yyyy = int(yyyy)
+                self.mm = int(mm)
+                self.dd = int(dd)
         except (ValueError, TypeError):
-            raise ValueError('The API version {} is not in '
-                             'YYYY-MM-DD or YYYY-MM-DD-preview format'.format(api_version_str))
+            raise ValueError('The API version {} is not in a '
+                             'supported format'.format(api_version_str))
 
     def __eq__(self, other):
-        return self.yyyy == other.yyyy and self.mm == other.mm and \
+        return self.latest == other.latest and self.yyyy == other.yyyy and self.mm == other.mm and \
```

Comment:
I use https://docs.python.org/2/library/functools.html#functools.total_ordering which means I only need to specify `__eq__` and one other method. Based on those two, it can work out all the other operations.